### ISSUE TITLE
how to install for ubuntu 18.04

### ISSUE THREAD

**Issue by manivara507** [NVIDIA/nvidia-docker#1250]
_The template below is mostly useful for bug reports and support questions. Feel free to remove anything which doesn't apply to you and add more information where it makes sense._

_Also, before reporting a new issue, please make sure that:_

- _You read carefully the [documentation and frequently asked questions](https://github.com/NVIDIA/nvidia-docker/wiki)._
- _You [searched](https://github.com/NVIDIA/nvidia-docker/issues?utf8=%E2%9C%93&q=is%3Aissue) for a similar issue and this is not a duplicate of an existing one._
- _This issue is not related to [NGC](https://github.com/NVIDIA/nvidia-docker/wiki/NGC), otherwise, please use the [devtalk forums](https://devtalk.nvidia.com/default/board/200/nvidia-gpu-cloud-ngc-users/) instead._
- _You went through the [troubleshooting](https://github.com/NVIDIA/nvidia-docker/wiki/Troubleshooting) steps._

---

### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)

 - [ ] Some nvidia-container information: `nvidia-container-cli -k -d /dev/tty info`
 - [ ] Kernel version from `uname -a`
 - [ ] Any relevant kernel output lines from `dmesg`
 - [ ] Driver information from `nvidia-smi -a`
 - [ ] Docker version from `docker version`
 - [ ] NVIDIA packages version from `dpkg -l '*nvidia*'` _or_ `rpm -qa '*nvidia*'`
 - [ ] NVIDIA container library version from `nvidia-container-cli -V`
 - [ ] NVIDIA container library logs (see [troubleshooting](https://github.com/NVIDIA/nvidia-docker/wiki/Troubleshooting))
 - [ ] Docker command, image and tag used

**Comment by cyremur** [comment]
I just installed it last week and the steps in the Qucikstart section of the readme work. What problem did you encounter?

`# Add the package repositories
distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | sudo apt-key add -
curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.list | sudo tee /etc/apt/sources.list.d/nvidia-docker.list

sudo apt-get update && sudo apt-get install -y nvidia-container-toolkit
sudo systemctl restart docker`

**Comment by shayanjoya** [comment]
If you are getting this:
```
nvidia-container-toolkit : Depends: libnvidia-container-tools (>= 0.1.0) but it is not going to be installed
                            Depends: libnvidia-container-tools (< 2.0.0) but it is not going to be installed
 nvidia-docker : Depends: sysv-rc (>= 2.88dsf-24) but it is not installable or
                          file-rc (>= 0.8.16) but it is not installable
```
when you try to run ```sudo apt-get update && sudo apt-get install -y nvidia-container-toolkit```
I found after running apt-get update a missing PUBKEY. You can fix it by following https://chrisjean.com/fix-apt-get-update-the-following-signatures-couldnt-be-verified-because-the-public-key-is-not-available/

**Comment by qhaas** [comment]
Just uninstalled / reinstalled the nvidia-docker related packages from the upstream repo in x86-64 ubuntu 18.04 by following the quick start guide (as summarized by @cyremur ) and it worked with docker 19.03.  Here are the packages I have installed:
```
apt list --installed | grep container

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

containerd.io/now 1.2.10-3 amd64 [installed,local]
libnvidia-container-tools/bionic,now 1.0.7-1 amd64 [installed,automatic]
libnvidia-container1/bionic,now 1.0.7-1 amd64 [installed,automatic]
nvidia-container-toolkit/bionic,now 1.0.5-1 amd64 [installed]
```

The story is likely different on some architectures with dated packages (e.g. ppc64le) and podman, but x86-64 with docker seems to be fine.

**Comment by klueska** [comment]
We recently updated our build system to support cross-builds on the following platforms:
```
+----------------------+-------------+----------------+---------+-----------------+
|  OS Name / Version   |  Identifier | amd64 / x86_64 | ppc64le | arm64 / aarch64 |
+======================+=============+================+=========+=================+
| Amazon Linux 1       | amzn1       |       X        |         |                 |
| Amazon Linux 2       | amzn2       |       X        |         |                 |
| Amazon Linux 2017.09 | amzn2017.09 |       X        |         |                 |
| Amazon Linux 2018.03 | amzn2018.03 |       X        |         |                 |
| Open Suse Leap 15.0  | sles15.0    |       X        |         |                 |
| Open Suse Leap 15.1  | sles15.1    |       X        |         |                 |
| Debian Linux 9       | debian9     |       X        |         |                 |
| Debian Linux 10      | debian10    |       X        |         |                 |
| Centos 7             | centos7     |       X        |    X    |                 |
| Centos 8             | centos8     |       X        |    X    |        X        |
| RHEL 7.4             | rhel7.4     |       X        |    X    |                 |
| RHEL 7.5             | rhel7.5     |       X        |    X    |                 |
| RHEL 7.6             | rhel7.6     |       X        |    X    |                 |
| RHEL 7.7             | rhel7.7     |       X        |    X    |                 |
| RHEL 8.0             | rhel8.0     |       X        |    X    |        X        |
| RHEL 8.1             | rhel8.1     |       X        |    X    |        X        |
| RHEL 8.2             | rhel8.2     |       X        |    X    |        X        |
| Ubuntu 16.04         | ubuntu16.04 |       X        |    X    |                 |
| Ubuntu 18.04         | ubuntu18.04 |       X        |    X    |        X        |
| Ubuntu 19.04         | ubuntu19.04 |       X        |    X    |        X        |
| Ubuntu 19.10         | ubuntu19.10 |       X        |    X    |        X        |
| Ubuntu 20.04         | ubuntu20.04 |       X        |    X    |        X        |
+----------------------+-------------+----------------+---------+-----------------+
```

The documentation hasn't been updated to explain how to do this yet, but it basically boils down to two things:

1) Enable `experimental` features in docker, which boils down to adding the following to your `/etc/docker/daemon.json` file:
```
{
"experimental": true
}
```
2) Install `binfmt-support` via `apt-get` or `yum` and invoke the following command:
```
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes -c yes
```

Once this is done, you should be able to invoke any of the new `make` targets that allow building on `amd64`, `arm64`, and `ppc64le` on any platform.
